### PR TITLE
docs: fix a misleading sentence in component class testing section

### LINKED ISSUE #36155: Docs: Testing section: component-class-testing, a little mistake
# 📚 Docs or angular.io bug report

### Description

There is a misleading sentence in the component class testing section, precisely the following sentence: This component class has no dependencies. To test a service with no dependencies, you create it...

Here, we jump from testing a component to testing a service.

## 🔬 Minimal Reproduction

This can be checked here: https://angular.io/guide/testing#component-class-testing

### What's the affected URL?**
https://angular.io/guide/testing#component-class-testing

## 🔥 Exception or Error
<pre><code>

</code></pre>

### PR DESCRIPTION
**Pull request by Fouzyyyy**
Docs: Testing -> Component class testing, a misleading sentence.

Fixes #36155

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

The current sentence in the Component class testing section looks like: 

This component class has no dependencies. To test a _service_ with no dependencies...

Issue Number: #36155

## What is the new behavior?

The new sentence will look like:

Just like services without dependencies, you create a component...

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by Fouzyyyy** [comment]
@googlebot I signed it!

**Comment by Fouzyyyy** [comment]
@sonukapoor sorry for bothering, what should I do in this case? `lint` failing, reason: `API rate limit exceeded`. Someone else should trigger the pipeline?

**Comment by sonukapoor** [comment]
@Fouzyyyy That seems to be an issue with CircleCI. @gkalpak / @petebacondarwin  can you help with this?

File: aio/content/guide/testing.md [hunk 1]
```diff
@@ -508,7 +508,7 @@ You might decide only to test that the `clicked()` method
 toggles the light's _on/off_ state and sets the message appropriately.
 
 This component class has no dependencies.
-To test a service with no dependencies, you create it with `new`, poke at its API,
+To test a component with no dependencies, you create it with `new`, poke at its API,
```
Comment (gkalpak) [inline aio/content/guide/testing.md]: AFAICT, `service` is intended here. What this paragraph is saying is that "this is how you would do it for services, you do the same for components (with no dependencies)". I accept this may be a little confusing. Feel free to suggest a different phrasing, but changing `service` to `component` here is not correct.

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx for submitting a PR, @Fouzyyyy. I don't think this change is correct (see my comment).
Don't worry about the CI failure for now.

**Comment by petebacondarwin** [comment]
The lint job failure was trying to check that the commit messages were valid. The error indicated that our CircleCi job had made too many unauthenticated requests to Github's API. I have restarted the job.

File: aio/content/guide/testing.md (hunk 1 above)
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: @gkalpak thanks a lot for your feed-back, it makes sense to maybe rephrase it to eliminate the misleading part of the sentence.

How/where can I suggest something else? Can I just do it here? Or at the issue level? Or simply push a change?

Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: I just pushed a new suggestion which eliminates a potential misunderstanding.

**Review by gkalpak — APPROVED** [review]
LGTM, thx ✨
I'll leave it for someone on the docs team give the final approval.

File: aio/content/guide/testing.md [hunk 2]
```diff
@@ -508,9 +508,8 @@ You might decide only to test that the `clicked()` method
 toggles the light's _on/off_ state and sets the message appropriately.
 
 This component class has no dependencies.
-To test a service with no dependencies, you create it with `new`, poke at its API,
+Just like services without dependencies, you create a component with `new`, poke at its API,
```
Comment (jbogarthyde) [inline aio/content/guide/testing.md]: As you do for services with no dependencies,

**Review by jbogarthyde — APPROVED** [review]
minor edit

**Review by jbogarthyde — APPROVED** [review]
LGTM

File: aio/content/guide/testing.md [hunk 3]
```diff
@@ -508,9 +508,8 @@ You might decide only to test that the `clicked()` method
 toggles the light's _on/off_ state and sets the message appropriately.
 
 This component class has no dependencies.
-To test a service with no dependencies, you create it with `new`, poke at its API,
+As you do for services with no dependencies, you create a component with `new`, poke at its API,
```
[the remark this answers is not in the document]
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: Yeah, more concise.

[the remark this answers is not in the document]
Comment (gkalpak) [reply aio/content/guide/testing.md]: Sorry, but I don't think the sentence makes sense now :grin:

[the remark this answers is not in the document]
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: Sorry @aikidave  actually the goal here is to refer to the services for the sake of giving an example, because here we are speaking about components without dependencies.

[the remark this answers is not in the document]
Comment (aikidave) [reply aio/content/guide/testing.md]: "As you do" implies that this is knowledge the user already has. Are you sure that's the case? If not, you should at least link to testing components that do not have dependencies is discussed.

From my reading of the topic, this content appears right at the beginning. Maybe what you should say is:

To test a service that has no dependencies, you create a component with `new`...

[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/testing.md]: I think you were right the first time - the service-with-no-dependencies case is discussed above, the component-with-no-dependencies uses the same method. So this is correct.

[the remark this answers is not in the document]
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: We may use: As for services with no dependencies, you create a component...

Like this, we eliminate the assumption that the reader knows or not, that a service with no dependencies could be created with the `new` keyword.

[the remark this answers is not in the document]
Comment (aikidave) [reply aio/content/guide/testing.md]: Again, though, the phrase "as for" indicates that you're talking about something that you already brought up. That's not the case here.

Yet another option: "For services that have no dependencies, you create a component..."

[the remark this answers is not in the document]
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: "For services that have no dependencies, you create a component..." sorry, does this make sense to you? jumping from a service to a component in the same sentence?

[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/testing.md]: No, it does not make sense. Please use my suggestion instead. 
"Handle the case of a component with no dependencies in the same way that you handle a service with no dependencies. Create a component..."

[the remark this answers is not in the document]
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: All right, change is coming.

[the remark this answers is not in the document]
Comment (aikidave) [reply aio/content/guide/testing.md]: That's why I prefer my original wording. How does this read to you:

This component class has no dependencies. To test these types of classes, follow the same steps as you would for a service that has no dependencies:

1. Create a component using the `new` keyword.
2. Poke at its API.
3. Assert expectations on its public state.

**Review by jbogarthyde — CHANGES_REQUESTED** [review]
Please make the suggested change to make the comparison of services to components without dependencies more explicit

File: aio/content/guide/testing.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/testing.md]: Yes, that works. Even more explicit.

[the remark this answers is not in the document]
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: hmm, it sounded to me a bit misleading, that's why I opened this MR, it may just be my perception. Should we then just leave it as is?

**Comment by Fouzyyyy** [comment]
> Please make the suggested change to make the comparison of services to components without dependencies more explicit

Which one do you refer to?

File: aio/content/guide/testing.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (aikidave) [reply aio/content/guide/testing.md]: What's misleading with my rewrite? Can you clarify?

[the remark this answers is not in the document]
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: I was referring to this sentence: 

To test a **service** that has no dependencies, you create a **component** with `new`...

It's just the fact of drifting from a service to a component in the same sentence.

**Review by jbogarthyde — APPROVED** [review]
That looks good to me.  Now it looks like you have to rebase on the latest master.

File: aio/content/guide/testing.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (aikidave) [reply aio/content/guide/testing.md]: Ah. Does my rewrite work better for you? Reposting for convenience:

This component class has no dependencies. To test these types of classes, follow the same steps as you would for a service that has no dependencies:

1. Create a component using the new keyword.
2. Poke at its API.
3. Assert expectations on its public state.

[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/testing.md]: This would also work for me, so my approval stands if you use this.

[the remark this answers is not in the document]
Comment (Fouzyyyy) [reply aio/content/guide/testing.md]: @aikidave thanks, yeah that one is way more clear to me as a consumer of the documentation, I'll push changes in a short moment.

**Review by aikithoughts — APPROVED** [review]
Thanks for taking the time to go through this!

**Comment by Fouzyyyy** [comment]
> That looks good to me. Now it looks like you have to rebase on the latest master.

Pushed the rewrite suggested by @aikidave . Also rebased on master.

**Comment by Fouzyyyy** [comment]
Is there anything left here?
